source/elements/oneDAL/source/data_management/index.rst

```diff
@@ -5,12 +5,140 @@
+
+Data are not always fit the local memory, especially when processing with
+accelerators. Data source provides an interface to load data by batches and
+orchestrates its efficient extraction directly into local memory of the device.
+Therefore data source enables complex scenarios of data analytics, like
+:term:`online computations <Online mode>`.
+
+Table builder
+-------------
+
+Some :term:`table` objects require complex procedures to create them. To make
+easy-of-use table creation from data of various :term:`data formats <Data
+format>` or other objects of different types, |dal_short_name| introduces a
+special :term:`builder` concept for table types. See :ref:`table builders
+<table_builders>` section for details.
+
+Accessor
+--------
+
+There are a lot of :term:`data formats <Data format>` used in :term:`datasets
+<Dataset>` to store the data. |dal_short_name| defines different types of
+:term:`tables <Table>`, table builders and other objects to support different
+logical types of data and various :term:`data formats <Data format>`. However,
+all these objects have different logic to access their underlying data.
+
+Accessor allows:
+
+- to make uniform access to the data from various sets of objects of different
+  types.
+
+- to perform efficient access to the data without exposing implementation
+  details of a data container.
+
+- to give an ability to focus on the data acquisition in the desired :term:`data
+  format` with a wanted set of operations over the data.
+
+See :ref:`accessors` section for details.
+
+Relations between concepts
+==========================
+
+Typical workflow with data in |dal_short_name| consists of next steps:
+
+1. Select an appropriate data source for extracting out-of-memory data, or
+   select a table builder to manually create a table object.
+
+2. Create :term:`table` object - a numerical representation of data, with use
+   of selected data source, table builder or table objetc constructor directly.
+
+3. Run a selected :ref:`algorithm <Algorithms>` with created table object.
+
+4. Get algorithm :ref:`results <Result>` and access their data directly or using
+   selected accessor.
+
+.. image:: _static/typical_data_flow.png
+  :width: 800
+  :alt: Typical data flow
+
+Next sequence diagram briefly shows dependencies between table, builder, and
+accessor concepts.
+
+Builder is responcible to create a table object, which data can be accessed
+either directly using table interface or with help of accessor object. Since the
+table object is immutable, this is not possible to change the data inside it.
+
+To change the data of table object, its data should be moved inside builder
+object. Once it done, data inside builder can be changed by the accessor.
+
+See sections about :ref:`tables`, :ref:`table builders <table_builders>`, and
+:ref:`accessors` for details.
+
+.. image:: _static/table_builder_accessor_sequence.png
+  :width: 800
+  :alt: Sequence diagram of accessor-builder-table relations
```

Comment:
Describe/discuss the content of the diagram so I'd be able to quickly understand what I have to implement for the concepts
The diagram contains colors. What do they mean
Also say about UML etc (the interactions between ... are described/represented with UML sequence diagram. It shows responsibilities, action, etc or similar

We will discuss things over the phone